✨ [extension] add columns to the event list

BenoitZugmeyer opened the pull request:
## Motivation

When adding a new field or investigating a specific field value, it is a bit annoying to expand it
on every event.

## Changes

* Use a grid instead of a table for event list (this was not strictly needed but I find it more predictable even if it has a few issues)
* Add support for columns in the event list to display some arbitrary fields.
  * Add columns by clicking on the json fields or by using the "Add column" button
  * Reorder columns by drag and dropping
  * Remove columns by dragging them out of the header

![Screenshot 2023-08-24 at 12 25 21](https://github.com/DataDog/browser-sdk/assets/61787/344a5042-0dbc-4a02-a644-65157f3c7476)

Note: see, the space between the facet list and the event list is back!

## Testing

<!-- How can the reviewer confirm these changes do what you say they do? Are there automated tests? -->

- [x] Local
- [ ] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

amortemousque commented on developer-extension/src/panel/components/tabs/eventsTab/eventsList.tsx:
```diff
@@ -1,19 +1,213 @@
 
-export function EventsList({ events, filters }: { events: SdkEvent[]; filters: EventFilters }) {
   return (
-    <Table striped verticalSpacing="xs">
-      <tbody>
```
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why move from a table to a grid?

BenoitZugmeyer replied:
I reverted back to a table, the grid offers more control over the exact size of columns, but was too much trouble to implement. I want to revisit this one day when I have time :)

bcaudan commented:
After playing with the UI:

🐛 You can "add column" on an already added column which create a new column with empty values

![Screenshot 2023-08-25 at 14 04 59](https://github.com/DataDog/browser-sdk/assets/1331991/ccfce562-74fc-41cb-9622-301549d5e4aa)
![Screenshot 2023-08-25 at 14 05 09](https://github.com/DataDog/browser-sdk/assets/1331991/dac86971-2345-4627-961e-13758575430d)

🐛 the display does not seem to handle well long URL in the message, maybe truncating them?

![Screenshot 2023-08-25 at 14 13 38](https://github.com/DataDog/browser-sdk/assets/1331991/1db1e4b7-ae15-435b-9d9f-3d16b4cc7b5d)


💬 for removing a column, drag and drop outside of the header seems quite unusual.
What about adding a dropdown on column header with remove column (move left / right could also be added)?  

💬 to indicate the columns can be reorderd by drag and drop, what about displaying "move" cursor on hover?

BenoitZugmeyer commented:
Current dependencies on/for this PR:

* main:
  * **PR #2369**
    * **PR #2370**
      * **PR #2371**
        * **PR #2372** 👈

This comment was autogenerated by Freephite.

BenoitZugmeyer commented:
> After playing with the UI:
> 
> 🐛 You can "add column" on an already added column which create a new column with empty values
> 

Fixed

> 
> 🐛 the display does not seem to handle well long URL in the message, maybe truncating them?
 
Fixed by reverting to using a Table, so now long strings should wrap properly

> 
> 💬 for removing a column, drag and drop outside of the header seems quite unusual. What about adding a dropdown on column header with remove column (move left / right could also be added)?

Added a button to remove columns.

> 
> 💬 to indicate the columns can be reorderd by drag and drop, what about displaying "move" cursor on hover?

Devtools tabs can be reordered by drag and drop, but don't have a "move" cursor. I think it could be a good idea to follow the same UI, so I didn't change the cursor. I will revisit the drag behavior later to make it closer to the native devtools experience.
